path/path.go

```diff
@@ -37,5 +45,55 @@ func (p Path) String() string {
 }
 
 func FromSegments(seg ...string) Path {
-	return Path(strings.Join(seg, "/"))
+	var pref string
+	if seg[0] == "ipfs" || seg[0] == "ipns" {
+		pref = "/"
+	}
+	return Path(pref + strings.Join(seg, "/"))
+}
+
+func ParsePath(txt string) (Path, error) {
+	parts := strings.Split(txt, "/")
+	if len(parts) == 1 {
+		kp, err := ParseKeyToPath(txt)
+		if err == nil {
+			return kp, nil
+		}
+	}
+	if len(parts) < 3 {
+		return "", ErrBadPath
+	}
```

Comment:
we should decide what constitutes a valid path.

Are these valid?

```
QmTcZtzoDaMaWhpS7YB2bZUJME4nEKpPChZBrfaeTpXc8c/foo/bar
foo/bar/baz
QmTcZtzoDaMaWhpS7YB2bZUJME4nEKpPChZBrfaeTpXc8c/foo
```
? `ParsePath` would say no, `FromSegments` would say yes.